File: src/lessons/welcome/methods/slug/SlugTrackingEntity.c
//RemoteBuggle

/* BEGIN TEMPLATE */
int isFacingTrail(Color c){

	/* BEGIN SOLUTION */
	if (isFacingWall())
		return 0;

	forward(1);
	int res = (getGroundColor() == c);
	backward(1);
	return res;

	/* END SOLUTION */
}
/* END TEMPLATE */


void hunt(Color c) {
	while (! isOverBaggle()) {
		if (isFacingTrail(c)) {
			brushDown();
			forward(1);
			brushUp();
		} else {
			left();
		}
	}
	pickupBaggle();
}
void run(){
	hunt(green);
}

File: src/lessons/welcome/methods/slug/SlugTrackingEntity.java
package lessons.welcome.methods.slug;

import java.awt.Color;

public class SlugTrackingEntity extends plm.universe.bugglequest.SimpleBuggle {

  @Override public void run()
  {
    while (!isOverBaggle()) {
      if (isFacingTrail()) {
        brushDown();
        stepForward();
        brushUp();
      } else {
        left();
      }
    }
    pickupBaggle();
  }

  /* BEGIN TEMPLATE */
  boolean isFacingTrail()
  {
    // Write your code here
    /* BEGIN SOLUTION */
    if (isFacingWall())
      return false;
    stepForward();
    boolean res = getGroundColor() == Color.green;
    stepBackward();
    return res;
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}
